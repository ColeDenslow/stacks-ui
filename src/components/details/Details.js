import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';

import './Details.css';

function Details(props) {
  const { className, textAlign, title, description, ...otherProps } = props;
  const classNames = cn([
    'stx-details',
    className,
    { 'stx-details--left': textAlign === 'left' },
    { 'stx-details--center': textAlign === 'center' },
    { 'stx-details--right': textAlign === 'right' },
  ]);

  return (
    <div
      className={classNames}
      {...otherProps}
    >
      <div
        className='stx-details__title'
      >
        {title}
      </div>
      <div
        className='stx-details__description'
      >
        {description}
      </div>
    </div>
  );
}

Details.defaultProps = {
  textAlign: 'left',
};

Details.propTypes = {
  className: PropTypes.string,
  description: PropTypes.string,
  title: PropTypes.string,
  textAlign: PropTypes.oneOf(['left', 'center', 'right']),
};

export default Details;
