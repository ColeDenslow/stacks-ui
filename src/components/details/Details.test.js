import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Details from './Details';

describe('Details', () => {
  test('Details renders with basic props', () => {
    const output = mount(<Details
      className='the-class'
      description='The description in the widget'
      title='The widget title'
    />);

    expect(toJson(output)).toMatchSnapshot();
  });

  test('Details renders with text align center', () => {
    const output = mount(<Details
      className='the-class'
      description='The description in the widget'
      title='The widget title'
      textAlign='center'
    />);

    expect(toJson(output)).toMatchSnapshot();
  });

  test('Details renders with text align right', () => {
    const output = mount(<Details
      className='the-class'
      description='The description in the widget'
      title='The widget title'
      textAlign='right'
    />);

    expect(toJson(output)).toMatchSnapshot();
  });
});
