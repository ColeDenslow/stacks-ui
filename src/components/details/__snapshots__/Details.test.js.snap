// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details Details renders with basic props 1`] = `
<Details
  className="the-class"
  description="The description in the widget"
  textAlign="left"
  title="The widget title"
>
  <div
    className="stx-details the-class stx-details--left"
  >
    <div
      className="stx-details__title"
    >
      The widget title
    </div>
    <div
      className="stx-details__description"
    >
      The description in the widget
    </div>
  </div>
</Details>
`;

exports[`Details Details renders with text align center 1`] = `
<Details
  className="the-class"
  description="The description in the widget"
  textAlign="center"
  title="The widget title"
>
  <div
    className="stx-details the-class stx-details--center"
  >
    <div
      className="stx-details__title"
    >
      The widget title
    </div>
    <div
      className="stx-details__description"
    >
      The description in the widget
    </div>
  </div>
</Details>
`;

exports[`Details Details renders with text align right 1`] = `
<Details
  className="the-class"
  description="The description in the widget"
  textAlign="right"
  title="The widget title"
>
  <div
    className="stx-details the-class stx-details--right"
  >
    <div
      className="stx-details__title"
    >
      The widget title
    </div>
    <div
      className="stx-details__description"
    >
      The description in the widget
    </div>
  </div>
</Details>
`;
